dg/utils/click.py

```diff
@@ -26,6 +26,49 @@
 import dg.utils.openshift
 
 
+def check_cloud_pak_for_data_options(
```

Comment:
If possible, the `--help` documentation of each  command should specify which values are required together. For example, when `use_dev` is active, artifactory credentials are necessary.

Apparently this can also be addressed by using the following extension for option groups: 
https://pypi.org/project/click-option-group/
https://github.com/click-contrib/click-option-group/

Not saying we necessarily need the plugin, but it should be reflected in the help text somehow, if possible.

Reply:
Click does not support option groups, which I do not understand as I think it is basic functionality. The PowerShell scripting language supports [parameter sets](https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/cmdlet-parameter-sets?view=powershell-7.1), which is exactly what I would like to have.

I already evaluated the suggested plug-in a while ago, but it does not fulfill our needs as you cannot group mutually exclusive option sets. Therefore, please come up with a concrete proposal on how to solve this problem.

Reply:
@jensmueller-com Handling mutually exclusive option sets is definitely supported, it's part of their self-coined 'Simple example' here: https://github.com/click-contrib/click-option-group/#a-simple-example

Other various behaviors can also be used https://click-option-group.readthedocs.io/en/latest/tutorial.html#behavior-and-relationship-among-options (If that isn't enough, GroupedOption can also be inherited from and an own handler can be written, but I don't think we need this.)

So to have a few more concrete proposals, picking either works to address the problem I think:
A) A simple solution, with an Option group not enforcing any relations between group options, which will basically only render `--help` text more cleanly. This would still use your check method in the following method body.

B) Using `RequiredMutuallyExclusiveOptionGroup` and only keeping the "main flag" of the option group as required, so `use_dev` or `ibm_cloud_pak_for_data_entitlement_key`. This would still use your check method in the following method body.

C) Implementing our own GroupedOption class, that also handles existing context information, so this doesn't have to be deferred to a separate method call in the option logic, but will be implicitly checked. This would move the option check logic to the newly created GroupedOption type.
![image](https://user-images.githubusercontent.com/62256845/101920907-ac98e300-3bcc-11eb-95b7-19f49eb88f63.png)

I think C would probably be most re-usable especially considering it could have a lot of the logic baked in we currently have for context handling, but for the sake of UX and readability the other two would also be sufficient.